src/app/modules/cerf/cerf.component.ts

```diff
@@ -239,6 +239,37 @@ export class CerfComponent {
 	isLabelActive(label: string): boolean {
 		return this.categoriesActive.includes(label);
 	}
+	
+	//sets existing label to an active one marked on the cerf
+	toggleLabel(label: string): void {
+		let index = this.categoriesActive.indexOf(label);
+		if(index == -1) { 	//doesn't exist as an active category
+			this.categoriesActive.push(label);
+		} else { 	//removes the given lable as an active category
+			this.categoriesActive.splice(index);
+		}
+	}
+	
+	//creates new label to be used for tags on cerfs
+	newLabel(label: string): void {
+		//label already exists check
+		if (!this.categoryButtons.includes(label)) {
+			this.categoryButtons.push(label);
+		}
+		this.addingCategory = false;
+	}
+
+	//deletes existing labels from the array of categories
+	removeLabel(index: number): void {
+		this.categoryButtons.splice(index);
```

Comment:
I think the rest of this looks good. But I just noticed you're using splicing to remove elements of an array. Won't this cut off everything after the index as well? So like, if you had [0,1,2,3,4,5] and splice off index 2, it'd just be [0,1]?

Reply:
in the function above this i think you use splicing as well, should make sure that one behaves alright too